Den 3147 Reading and writing a serial number to the Orion board's EEPROM

CraigDensity opened the pull request:
Added functions to read and write a serial number to the FX2's I2C connected EEPROM on the Orion board.

CraigDensity committed:
Added commands to store and read a serial number on the Orion board's EEPROM.
DEN-3147

bradfordbarr commented on cycfx2prog-0.47/cycfx2dev.h:
```diff
@@ -34,6 +34,8 @@ class CypressFX2Device
 		int _ProgramIHexLine(const char *buf,const char *path,int line);
+                // Internally used to calculate a serial number's checksum
+                unsigned short Computecrc16(const unsigned char* data_p, unsigned char length);
 	private:
```
Why are there two `private` labels?

CraigDensity replied:
Good Point. I'll remove it

CraigDensity committed:
Removing second Private label bases on comments from DEN-3147 review.

ek9852 commented on cycfx2prog-0.47/cycfx2dev.cc:
```diff
@@ -1069,6 +1069,150 @@ int CypressFX2Device::SerialNumberWrite(const unsigned char *ctl_buf, size_t ctl
         return(rv);
 }
 
+int CypressFX2Device::OrionSerialNumberRead(const unsigned char *ctl_buf, size_t ctl_buf_size)
```
const unsigned char => unsigned char ?

ek9852 commented on cycfx2prog-0.47/cycfx2dev.h:
```diff
@@ -124,6 +125,12 @@ class CypressFX2Device
                 int SerialNumberWrite(const unsigned char *ctl_buf,
                         size_t ctl_buf_size);
+                // Read the serial number from the Orion board
+                int OrionSerialNumberRead(const unsigned char *ctl_buf,
```
const unsigned char => unsigned char ?

ek9852 commented on cycfx2prog-0.47/cycfx2prog.cc:
```diff
@@ -712,7 +716,50 @@ int main(int argc,char **arg)
+                else if(!strcmp(cmd,"setorionserial"))
+                {
+                    // if a serial number string has been provided and no more than 32 characters
+                    if(a[0] && *a[0] && (strlen(a[0]) < 33))
```
Not sure why it is 32 characters ? doesn't we write ascii 64 characters directly ?
Or we write binary 32 characters ?

CraigDensity replied:
Good Catch! Thanks Keith!

CraigDensity committed:
Increased the allowed orion serial number size to 64 characters at the command line, up from 32.
Removed const from function header that read into the const buffer.
Changes based on pull review for Jira DEN-3147
